torch/optim/radam.py

```diff
@@ -357,6 +373,15 @@ def _multi_tensor_radam(
         # Update steps
         torch._foreach_add_(grouped_state_steps, 1)
 
+        if has_complex:
```

Comment:
Consider factoring this out into an aux function shared across optimisers

Reply:
Yes, I will look into it in following up to https://github.com/pytorch/pytorch/pull/110607. However, I'm not entirely sure there's a neat way to handle it due to users of the functional APIs. For instance, we already tried to factor out the improvements into the `_init_group`, which was a cleaner fix, but were worried that some people may use the functional APIs directly. https://github.com/pytorch/pytorch/pull/110596#pullrequestreview-1660058364.